ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -608,16 +614,55 @@ static void HandleKernelRequest(void* messageMemory, uint32_t messageSize)
         case MessageType_KtoU_NotifyDirectoryRenamed:
         case MessageType_KtoU_NotifyFileHardLinkCreated:
         {
+            // Walk up the directory tree and notify the provider about any directories
```

Comment:
I think we should keep the `HandleKernelRequest` function as pure routing code, and move this logic down into the other function that handles this type of message.

Reply:
Or a separate function for all of this logic here, since it's actually generating more callbacks to the provider